File: sdks/ts/src/sdk/models/operations/getfileversionretentions.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";


export enum GetFileVersionRetentionsDispositionActionEnum {
    PermanentlyDelete = "permanently_delete",
    RemoveRetention = "remove_retention"
}


export class GetFileVersionRetentionsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=disposition_action" })
  dispositionAction?: GetFileVersionRetentionsDispositionActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=disposition_after" })
  dispositionAfter?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=disposition_before" })
  dispositionBefore?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=file_id" })
  fileId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=file_version_id" })
  fileVersionId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=marker" })
  marker?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=policy_id" })
  policyId?: string;
}


export class GetFileVersionRetentionsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetFileVersionRetentionsQueryParams;
}


export class GetFileVersionRetentionsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  fileVersionRetentions?: shared.FileVersionRetentions;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/getfileversionretentions.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFileVersionRetentionsDispositionActionEnum string

const (
	GetFileVersionRetentionsDispositionActionEnumPermanentlyDelete GetFileVersionRetentionsDispositionActionEnum = "permanently_delete"
	GetFileVersionRetentionsDispositionActionEnumRemoveRetention   GetFileVersionRetentionsDispositionActionEnum = "remove_retention"
)

type GetFileVersionRetentionsQueryParams struct {
	DispositionAction *GetFileVersionRetentionsDispositionActionEnum `queryParam:"style=form,explode=true,name=disposition_action"`
	DispositionAfter  *string                                        `queryParam:"style=form,explode=true,name=disposition_after"`
	DispositionBefore *string                                        `queryParam:"style=form,explode=true,name=disposition_before"`
	FileID            *string                                        `queryParam:"style=form,explode=true,name=file_id"`
	FileVersionID     *string                                        `queryParam:"style=form,explode=true,name=file_version_id"`
	Limit             *int64                                         `queryParam:"style=form,explode=true,name=limit"`
	Marker            *string                                        `queryParam:"style=form,explode=true,name=marker"`
	PolicyID          *string                                        `queryParam:"style=form,explode=true,name=policy_id"`
}

type GetFileVersionRetentionsRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetFileVersionRetentionsQueryParams
}

type GetFileVersionRetentionsResponse struct {
	ClientError           *shared.ClientError
	ContentType           string
	FileVersionRetentions *shared.FileVersionRetentions
	StatusCode            int64
}
